core/storage/trie/impl/persistent_trie_batch_impl.cpp

```diff
@@ -81,15 +82,15 @@ namespace kagome::storage::trie {
 
   outcome::result<void> PersistentTrieBatchImpl::put(const Buffer &key,
                                                      const Buffer &value) {
-    return put(key, Buffer {value}); // would have to copy anyway
+    return put(key, Buffer{value});  // would have to copy anyway
   }
 
   outcome::result<void> PersistentTrieBatchImpl::put(const Buffer &key,
                                                      Buffer &&value) {
     bool is_new_entry = not trie_->contains(key);
-    auto res = trie_->put(key, std::move(value));
+    auto res = trie_->put(key, value);
```

Comment:
I guess in this case it's better to move it to the version with const Buffer&, that should save one copy.

Reply:
We must receive notifications from both versions of ```put```. So if we have guarantee the value will be inserted after 
```bool is_new_entry = not trie_->contains(key);``` then we can call notification before it will be put here
```auto res = trie_->put(key, value);```.

Reply:
I mean, move that code to that put and just call that put from this put.